src/project/mod.rs

```diff
@@ -549,6 +549,57 @@ impl Project {
         Ok(())
     }
 
+    pub fn remove_dependency(
+        &mut self,
+        name: impl AsRef<str>,
+        platform: Option<Platform>,
+    ) -> miette::Result<()> {
+        let searched: String;
+        let removed = match platform {
+            Some(p) => {
+                let t = get_toml_target_table(&mut self.doc, p, "dependencies")?;
+                searched = format!("target.{}.dependencies", p.as_str());
+                let removed = t.remove(name.as_ref());
+                removed
+            }
+            None => {
+                let dep_types = ["dependencies", "host-dependencies", "build-dependencies"];
+                let mut removed: Option<Item> = None;
+                for dep in dep_types {
+                    if let Item::Table(t) = &mut self.doc[dep] {
+                        if t.contains_key(name.as_ref()) {
+                            removed = t.remove(name.as_ref());
+                            break;
+                        }
+                    }
+                }
+
+                let mut tmp = dep_types
+                    .iter()
+                    .fold(String::from("["), |acc, v| acc + v + ", ");
+                tmp = tmp[..tmp.len() - 2].to_string();
+                tmp.push(']');
+                searched = tmp;
+
+                removed
+            }
+        };
+
+        if removed.is_some() {
+            self.save()?;
+            self.reload()?;
```

Comment:
This is a pretty ugly hack. I would prefer if you just removed the fields from the corresponding tables using the logic above.